com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/handlers/v3/ModuleSorter.java

```diff
@@ -0,0 +1,80 @@
+package com.sap.cloud.lm.sl.mta.handlers.v3;
+
+import static com.sap.cloud.lm.sl.common.util.CommonUtil.cast;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import com.sap.cloud.lm.sl.mta.builders.v2.ModuleDependenciesCollector;
+import com.sap.cloud.lm.sl.mta.model.v1.Module;
+import com.sap.cloud.lm.sl.mta.model.v3.DeploymentDescriptor;
+
+public class ModuleSorter extends com.sap.cloud.lm.sl.mta.handlers.v2.ModuleSorter {
+
+    String parallelDeploymentsProperty;
+
+    public ModuleSorter(DeploymentDescriptor descriptor, DescriptorHandler handler, String dependencyTypeProperty,
+        String hardDependencyType, String parallelDeploymentsProperty) {
+        super(descriptor, handler, dependencyTypeProperty, hardDependencyType);
+        this.parallelDeploymentsProperty = parallelDeploymentsProperty;
+    }
+
+    @Override
+    public List<? extends Module> sort() {
+        DeploymentDescriptor descriptorV3 = cast(descriptor);
+        if (!isParallelDeploymentsEnabled(descriptorV3) && !hasDeployedAfterAttribute(descriptorV3)) {
```

Comment:
In order to remove this "beautiful" if from here, could you add some object which defines the "sorting alghoritm".
For instance, based on some condition you could decide which method to use for sorting. I know it is complicated bu it will remove this complexity here which leads to a lot of misunderstandings :)